pkg/scte35/stream.go

```diff
@@ -0,0 +1,401 @@
+// Copyright 2022 Adrian of Doom
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or   implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+
+package scte35
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+// pktSz is the size of an MPEG-TS packet in bytes.
+const pktSz = 188
+
+// bufSz is the size of a read when parsing files.
+const bufSz = 13000 * pktSz
+
+// isIn16 is a test for slice membership
+func isIn16(slice []uint16, val uint16) bool {
```

Comment:
Re-organize this slightly... exported functions and structs on top, un-exported functions and structs below

Reply:
sounds reasonable.